spec/connect/api_spec.rb

```diff
@@ -272,24 +272,27 @@
   end
 
   describe "#system_migrations" do
-    shared_examples 'a query with no specified target base that returns migration paths' do |kind|
+    shared_examples 'a query with no specified target base that returns migration paths in the correct order' do |kind|
       before { stub_system_migrations_call(kind) }
 
       let(:products) do
         [
-          Remote::Product.new(identifier: 'SLES', version: '12', arch: 'x86_64', release_type: 'HP-CNB'),
-          Remote::Product.new(identifier: 'SUSE-Cloud', version: '7', arch: 'x86_64', release_type: nil)
+          Remote::Product.new(identifier: 'SLES', version: '12', arch: 'x86_64', release_type: 'HP-CNB')
         ]
       end
 
       let(:query) { { installed_products: products.map(&:to_params) } }
 
       its(:code) { is_expected.to eq 200 }
       its(:body) do
-        is_expected.to match_array([[
-          { 'identifier' => 'SLES', 'version' => '12.1', 'arch' => 'x86_64', 'release_type' => 'HP-CNB' },
-          { 'identifier' => 'SUSE-Cloud', 'version' => '8', 'arch' => 'x86_64', 'release_type' => nil }
-        ]])
+        is_expected.to match_array(
+          [
+            [{ 'identifier' => 'SLES', 'version' => '12.2', 'arch' => 'x86_64', 'release_type' => 'HP-CNB' }],
+            [{ 'identifier' => 'SLES', 'version' => '12.1', 'arch' => 'x86_64', 'release_type' => 'HP-CNB' }],
+            [{ 'identifier' => 'SLE-HPC', 'version' => '12.2', 'arch' => 'x86_64', 'release_type' => nil }],
+            [{ 'identifier' => 'SLE-HPC', 'version' => '12.1', 'arch' => 'x86_64', 'release_type' => nil }]
+          ]
+        )
```

Comment:
Do you think we might need a counter test, I mean a test that shows that SLES comes first when SLES is the base, and another test that shows that HPC comes first when HPC is base?

Reply:
nice idea! :+1: added a spec for that